README.md

```diff
@@ -1,9 +1,65 @@
+
+As stated in the Related Work section of [the OpenPubkey paper](https://eprint.iacr.org/2023/296.pdf):
+
+> “Sigstore [31, 42] is an open source project for signing and verifying software artifacts. Users can sign under their OpenID Connect identity by using the sigstore Fulcio Certificate Authority [43] which uses an immutable log to store a mapping between an OpenID Connect ID Token and a short lived public key enabling parties to attribute signatures to identities. The Fulcio CA (Certificate Authority) is trusted to create this mapping between an ID Token and a public key. Using OpenPubkey this trust can be eliminated as OpenPubkey does not need a trusted party to map ID Tokens to public keys. The Fulcio CA could in turn help OpenPubkey by acting as a public OpenPubkey verifier and OP public key database."
+
+### How does OpenPubkey ensure the nonce claim functions as nonce?
+
+In the user-identity scenario the CIC that contains the user's public key is hashed to the `nonce` claim in the ID Token. As OIDC requires that this field never repeat, OpenPubkey includes a random value, rz, in the CIC. Thus the hash of the CIC is always different and random. This maintains the required properties needed by the `nonce` claim in OIDC.
+
+```
+rz = crypto.random()
+CIC = {'rz': crypto.random(), 'upk': <publickey>, 'alg': 'EC256'}
+IDToken.nonce = SHA3(CIC)
+```
+
+### Does OpenPubkey present a privacy leak?
+
+The PK Tokens used by OpenPubkey contain the claims from the OIDC ID Tokens of the signer, so making them public, necessarily makes those claims public too. This may include elements of the signer’s identity such as the signer’s name or email addresses. It is up to users of OpenPubkey as to whether or not PK Tokens are made public.
+
+In a public artifact signing scenario, it could be argued that these very claims are the very claims upon which trust in the artifact should be based. However, some OIDC providers may include claims that the signer may wish to keep private. Users of OpenPubkey should consider carefully which OIDC providers to integrate with.
+
+Most OpenID Providers (OP) allow you to scope the fields in the ID Tokens. For instance Google’s OP is by default scoped to only include userinfo-email claims: name, email address and icon. Given that purpose of OpenPubkey is to enable parties to verify that a particular identity e.g., ethan@bastion.com, produced a particular signature, if you do not want signatures to be associated with OIDC identities, then OpenPubkey may not be a good fit for your use case.
+
+In other OpenPubkey deployment scenarios, such as those employed by BastionZero, the ID Tokens are not made publicly available.
```

Comment:
Is it worth mentioning something there is no information leakage in the workflow-identity use case?

Reply:
I want to address this more completely in the future because it really depends on what is meant by information leakage and the harm it causes. Generally the information in the workload-identity ID Token does not represent any sort of privacy impact but someone can always come up with an absurd counter example.

I'm going to invent an absurd scenario:

Someone could use OpenPubkey for a private repo in which they want to keep the very existence of the repo secret, but they want sign the repo and post the signature and PK Token on a public transparency log. Technically this would be an information leakage. If they want to keep the very existence of their repo secret, I would argue using OpenPubkey with a public transparency log isn't a great fit.

I think this concern requires a longer treatment than I can do right now.